### PR TITLE
Fixed #1087 : Login dialog now is destroyed when it is closed

### LINKED ISSUE #1087: Login credentials are not cleared in the viewer page
1. go to QA
1. open a map (Unesco Italian Items MS2)
1. try to log with username "pizza" password "pazza"
1. it should display a message error
1. open again the login form and the username and password are still there, they aren't cleared

this is not happening in the homepage

### PR DESCRIPTION
**Pull request by MV88**
Refactored the LoginModal, PasswordResetModal and UserDetailsModal components. now they are dialog by default and they are destroyed every time they are closed.

### CODE REVIEW

File: web/client/components/security/forms/PasswordReset.jsx [hunk 1]
```diff
@@ -68,7 +61,7 @@ const PasswordReset = React.createClass({
           return null;
       }
       let pw = this.refs.password.getValue();
-      if (pw !== null && pw.length < this.props.minPasswordSize) {
+      if (pw !== null && pw.length < this.props.minPasswordSize && pw.length > 0) {
           return <Alert bsStyle="danger"><Message msgId="user.passwordMinlenght" msgParams={{minSize: this.props.minPasswordSize}}/></Alert>;
```
Comment (MV88) [inline web/client/components/security/forms/PasswordReset.jsx]: i think is better to show the message when the user starts inserting something, or we can change the class from danger to info

File: web/client/localConfig.json [hunk 2]
```diff
@@ -733,8 +733,25 @@
...
+								"buttonSize": "small",
+								"includeCloseButton": false,
+								"useModal": false,
+								"closeGlyph": "1-close"
+							}, {
+								"buttonSize": "small",
+								"includeCloseButton": false,
+								"useModal": false,
+								"closeGlyph": "1-close"
+					 }]
```
Comment (MV88) [inline web/client/localConfig.json]: had to change the localConfig to get dialogs also in the home, (we can just return this renderDialog() in the components)

File: web/client/translations/data.fr-FR [hunk 3]
```diff
@@ -430,6 +430,7 @@
             "login": "Login",
             "logout": "Logout",
             "info": " Informations du compte",
+            "details": " User Details",
```
Comment (offtherailz) [inline web/client/translations/data.fr-FR]: Détails de l'utilisateur

Comment (MV88) [inline web/client/translations/data.fr-FR]: ok

**Pull request merged**